locator/topology.cc

```diff
@@ -63,19 +63,51 @@ std::string node::to_string(node::state s) {
 
 future<> topology::clear_gently() noexcept {
     _this_node = nullptr;
-    co_await utils::clear_gently(_datacenters);
+    // Keep the default dc and rack even if empty
+    // since _default_location points to them
+    for (auto dc_it = _datacenters.begin(); dc_it != _datacenters.end(); ) {
+        auto* dc = const_cast<datacenter*>(dc_it->second.get());
+        for (auto rack_it = dc->racks().begin(); rack_it != dc->racks().end(); ) {
+            auto* rack = const_cast<locator::rack*>(rack_it->second.get());
```

Comment:
Why these const casts? clear_gently isn't const so should have mutable access to everything.

Reply:
The const_cast:s are needed because the `_datacenters` and `dc->_racks` maps use `const datacenter*` and `const rack*` values, respectively, on purpose, for safety reasons.